modules/plugin/svg/src/main/java/org/geotools/renderer/style/SVGGraphicFactory.java

```diff
@@ -85,6 +99,56 @@
         formats.add("image/svg+xml");
     }
 
+    /** Hints we care about */
+    final private Map<Key, Object> implementationHints = new HashMap<>();
+    
+    private EntityResolver resolver;
+    
+    public SVGGraphicFactory(){
+        this( null );
+    }
+    
+    public SVGGraphicFactory(Map<Key, Object> hints){
+        if( hints != null && hints.containsKey(Hints.ENTITY_RESOLVER)){
+            // use entity resolver provided (even if null)
+            this.resolver = (EntityResolver) hints.get(Hints.ENTITY_RESOLVER);
+            this.implementationHints.put( Hints.ENTITY_RESOLVER, this.resolver );
+
+            if( this.resolver == null ){ // use null instance rather than check each time
+                this.resolver = NullEntityResolver.INSTANCE;
+            }
+        }
+        else {
+            this.resolver = defaultResolver();
+        }
+    }
+    private static final Logger LOGGER = org.geotools.util.logging.Logging.getLogger("org.geotools.renderer.style.SVGGraphicFactory");
+    protected static EntityResolver defaultResolver() {
```

Comment:
Not sure if this should be here - look like generic code useful to anyone looking for a resolver.  Perhaps put somewhere else?